src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/AvoidUninstantiatedInternalClasses.cs

```diff
@@ -44,7 +44,7 @@ public override void Initialize(AnalysisContext analysisContext)
 
             analysisContext.RegisterCompilationStartAction(startContext =>
             {
-                var instantiatedTypes = new ConcurrentDictionary<INamedTypeSymbol, object?>();
+                ConcurrentDictionary<INamedTypeSymbol, object?> instantiatedTypes = new ConcurrentDictionary<INamedTypeSymbol, object?>();
```

Comment:
Those explicit types were not necessary, are they?

Reply:
The commit introducing these changes is named "Use explicit type where compiler fails to evaluate nullability" so I think they matters.

Reply:
@jcouv yes, they are necessary. `var` doesn't work with local functions, at least not here.

Reply:
Do we need to file a compiler bug?

Reply:
I think there are already bugs filed for improving local function support. It hit Roslyn too.

Reply:
➡️ https://github.com/dotnet/roslyn/issues/41689 and https://github.com/dotnet/roslyn/issues/41705